compiler/AST/expr.cpp

```diff
@@ -3722,9 +3722,67 @@ CallExpr::insertAtTail(BaseAST* ast) {
 }
 
 
-FnSymbol* CallExpr::isResolved(void) {
-  SymExpr* base = toSymExpr(baseExpr);
-  return base ? toFnSymbol(base->var) : NULL;
+bool CallExpr::isEmpty() const {
+  return primitive == NULL && baseExpr == NULL;
+}
+
+
+// MDN 2016/01/29: This will become a predicate
+FnSymbol* CallExpr::isResolved() const {
+  return resolvedFunction();
+}
+
+
+FnSymbol* CallExpr::resolvedFunction() const {
```

Comment:
I'm not worried about algorithmic cost of the repeated work, more maintenance cost and the view that going forward it seems like you want that logic in resolvedFunction() rather than outside it.  Removing any potential duplication would make it more likely that additional checks would be inserted inside the method (but it sounds like you already want to do some clean up so I guess I'm agreeing with you?)